sunpy/net/base_client.py

```diff
@@ -266,19 +266,19 @@ def __repr__(self):
         """
         Returns the normal repr plus the pretty client __str__.
         """
-        return object.__repr__(self) + "\n" + str(self)
+        return _print_client(visible_entries=15, client=self)
 
     def __str__(self):
         """
         This enables the "pretty" printing of BaseClient.
         """
-        return _print_client(self)
+        return _print_client(visible_entries=-1, client=self)
```

Comment:
Should this not be None?

Reply:
I used list splicing for limiting rows. -1 index just gives the last element of the list. So that it prints the entire list.

Reply:
I think this is only correct if you index the list directly aka alist[-1]. If you slice it, it excludes the last element.

Reply:
My bad. I'll just add an if statement for that block.